File: 4-curso/2 cuatrimestre/DS/Practica 1/Ejercicios/Ejercicios 2/ejercicio2s2/src/main/java/com/p1/CadenaFiltros.java
package com.p1;

import java.util.ArrayList;

public class CadenaFiltros {
    private ArrayList<Filtro> filtros;

    CadenaFiltros(){
        filtros = new ArrayList<>();
    }

    CadenaFiltros(ArrayList<Filtro> f){
        filtros = f;
    }

     public String ejecutar(String texto){
        // Recorre cada filtro aplicándolo a la cadena
        for(Filtro filtro : filtros)
            // Se aplica cada filtro a la cadena y ésta reemplaza la original
            texto = filtro.ejecutar(texto);

        // Devuelve la cadena procesada
        return texto;
    }

    public void aniadirFiltro(Filtro filtro){
        filtros.add(filtro);
    }
}

File: 4-curso/2 cuatrimestre/DS/Practica 2/ejercicio_filtro/lib/CadenaFiltros.dart
import 'Filtro.dart';

class CadenaFiltros {
  late List<Filtro> _filtros;

  CadenaFiltros() {
    _filtros = [];
  }

  CadenaFiltros.n(List<Filtro> f) {
    _filtros = f;
  }

  String ejecutar(String texto) {
    for (Filtro filtro in _filtros) texto = filtro.ejecutar(texto);

    return texto;
  }

  List<Filtro> getFiltros() {
    return _filtros;
  }

  void aniadeFiltro(Filtro filtro) {
    _filtros.add(filtro);
  }

  void resetCadena() {
    _filtros = [];
  }
}
